### PR TITLE
chore: more squeezed simps

### PR DESCRIPTION
**Pull request by adomani**
One more PR with `simp`s squeezed thanks to #11246.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/SpecificGroups/Coxeter.lean [hunk 1]
```diff
@@ -458,7 +457,7 @@ def H₄ : Matrix (Fin 4) (Fin 4) ℕ :=
      2, 2, 5, 1]
 
 theorem H₄IsCoxeter : IsCoxeter H₄ where
-  symmetric := by simp [Matrix.IsSymm]; decide
+  symmetric := by unfold Matrix.IsSymm; decide
```
Comment (alexjbest) [inline Mathlib/GroupTheory/SpecificGroups/Coxeter.lean:460]: I guess these could have been simp with decide turned on instead? I'm a bit wary of these. Decide is a good example of a tactic that doesn't card too much what the goal is before but might run a lot slower if the input is in a different form

File: Mathlib/Algebra/Category/ModuleCat/Subobject.lean [hunk 2]
```diff
@@ -45,7 +45,8 @@ noncomputable def subobjectModule : Subobject M ≃o Submodule R M :=
         · apply LinearEquiv.toModuleIso'Left
           apply LinearEquiv.ofBijective (LinearMap.codRestrict (LinearMap.range S.arrow) S.arrow _)
           constructor
-          · simp [← LinearMap.ker_eq_bot, LinearMap.ker_codRestrict]
+          · simp only [← LinearMap.ker_eq_bot, LinearMap.mem_range, exists_apply_eq_apply,
+            forall_const, LinearMap.ker_codRestrict]
             rw [ker_eq_bot_of_mono]
```
Comment (YaelDillies) [inline Mathlib/Algebra/Category/ModuleCat/Subobject.lean:50]: Does this work?
```suggestion
          · simp [← LinearMap.ker_eq_bot, LinearMap.ker_codRestrict, ker_eq_bot_of_mono]
```

File: Mathlib/Algebra/FreeAlgebra.lean [hunk 3]
```diff
@@ -247,8 +247,8 @@ instance : Semiring (FreeAlgebra R X) where
   __ := instAddCommMonoid R X
   __ := instDistrib R X
   natCast n := Quot.mk _ (n : R)
-  natCast_zero := by simp; rfl
-  natCast_succ n := by simp; exact Quot.sound Rel.add_scalar
+  natCast_zero := by simp only [Nat.cast_zero]; rfl
```
Comment (YaelDillies) [inline Mathlib/Algebra/FreeAlgebra.lean:250]:
```suggestion
  natCast_zero := by simpa using rfl
```

File: Mathlib/Data/MvPolynomial/Monad.lean [hunk 4]
```diff
@@ -413,8 +413,9 @@ instance lawfulMonad : LawfulMonad fun σ => MvPolynomial σ R
     where
   pure_bind := by intros; simp [pure, bind]
   bind_assoc := by intros; simp [bind, ← bind₁_comp_bind₁]
-  seqLeft_eq := by intros; simp [SeqLeft.seqLeft, Seq.seq, (· <$> ·), bind₁_rename]; rfl
-  seqRight_eq := by intros; simp [SeqRight.seqRight, Seq.seq, (· <$> ·), bind₁_rename]; rfl
+  seqLeft_eq := by intros; simp only [SeqLeft.seqLeft, Seq.seq, (· <$> ·), bind₁_rename]; rfl
+  seqRight_eq := by intros; simp only [Seq.seq, (· <$> ·), bind₁_rename, Function.comp_const,
+                              rename_id]; rfl
```
Comment (YaelDillies) [inline Mathlib/Data/MvPolynomial/Monad.lean:418]: Can you try `simpa ... using rfl` here?

File: Mathlib/Data/Polynomial/Basic.lean [hunk 5]
```diff
@@ -430,9 +430,9 @@ theorem card_support_eq_zero : p.support.card = 0 ↔ p = 0 := by simp
 def monomial (n : ℕ) : R →ₗ[R] R[X] where
   toFun t := ⟨Finsupp.single n t⟩
   -- porting note (#10745): was `simp`.
-  map_add' x y := by simp; rw [ofFinsupp_add]
+  map_add' x y := by simp_rw [Finsupp.single_add]; rw [ofFinsupp_add]
```
Comment (YaelDillies) [inline Mathlib/Data/Polynomial/Basic.lean:433]:
```suggestion
  map_add' x y := by simp [Finsupp.single_add, ofFinsupp_add]
```
maybe?

File: Mathlib/GroupTheory/SpecificGroups/Coxeter.lean [hunk 6]
```diff
@@ -421,7 +420,7 @@ def G₂ : Matrix (Fin 2) (Fin 2) ℕ :=
      6, 1]
 
 theorem G₂IsCoxeter : IsCoxeter G₂ where
-  symmetric := by simp [Matrix.IsSymm]; decide
+  symmetric := by unfold Matrix.IsSymm; decide
```
Comment (YaelDillies) [inline Mathlib/GroupTheory/SpecificGroups/Coxeter.lean:423]: If you add an instance for `DecidablePred Matrix.IsSymm` (which should be pretty trivial), you will be able to golf this even further to
```suggestion
  symmetric := by decide
```

File: Mathlib/LinearAlgebra/LinearIndependent.lean [hunk 7]
```diff
@@ -1285,12 +1285,10 @@ open Submodule
    (instead of a data containing type class) -/
 theorem mem_span_insert_exchange :
     x ∈ span K (insert y s) → x ∉ span K s → y ∈ span K (insert x s) := by
-  simp [mem_span_insert]
+  simp only [mem_span_insert, forall_exists_index, and_imp]
   rintro a z hz rfl h
   refine' ⟨a⁻¹, -a⁻¹ • z, smul_mem _ _ hz, _⟩
-  have a0 : a ≠ 0 := by
-    rintro rfl
-    simp_all
```
Comment (YaelDillies) [inline Mathlib/LinearAlgebra/LinearIndependent.lean:1293]: This is a false positive, right?

**Review by YaelDillies — COMMENTED** [review]
!bench

**Review by adomani — COMMENTED** [review]
Thanks for the reviews!  I am skeptical about adding the `Decidable` instance for `Matrix.IsSymm` in the same PR.

Also, these issues are still using the old linter.  I have not yet had the time to really think through the new one, though.

**Comment by adomani** [comment]
!bench

**Comment by joneugster** [comment]
Same as in #11281: I understood that the implementation of the "flexible linter" addresses this PR and therefore this can be closed.

Please reopen if that's not correct!

**Pull request closed without merging**